### PR TITLE
FileTarget - Write time details to InternalLogger when file is older than MaxArchiveDays

### PR DESCRIPTION
**Pull request by snakefoot**
Followup to  #3714

### CODE REVIEW

File: src/NLog/Targets/FileArchiveModes/FileArchiveModeBase.cs [hunk 1]
```diff
@@ -111,7 +112,7 @@ public virtual List<DateAndSequenceArchive> GetExistingArchiveFiles(string archi
             }
 
             if (existingArchiveFiles.Count > 1)
-                existingArchiveFiles.Sort(FileSortOrderComparison);
+                existingArchiveFiles.Sort((x,y) => FileSortOrderComparison(x, y));
```
Comment (304NotModified) [inline src/NLog/Targets/FileArchiveModes/FileArchiveModeBase.cs]: Why this change? Do you think it's more clear then?

File: src/NLog/Targets/FileArchiveModes/FileArchiveModeBase.cs [hunk 2]
```diff
@@ -188,8 +189,16 @@ private bool ShouldDeleteFile(DateAndSequenceArchive existingArchiveFile, int re
...
                 var fileDateUtc = existingArchiveFile.Date.Date.ToUniversalTime();
-                if (fileDateUtc > MaxAgeArchiveFileDate && (NLog.Time.TimeSource.Current.Time.Date.ToUniversalTime() - fileDateUtc).TotalDays > maxArchiveDays)
-                    return true;
+                if (fileDateUtc > MaxAgeArchiveFileDate)
+                {
+                    var currentDateUtc = NLog.Time.TimeSource.Current.Time.Date.ToUniversalTime();
+                    var fileAgeDays = (currentDateUtc - fileDateUtc).TotalDays;
+                    if (fileAgeDays > maxArchiveDays)
+                    {
+                        InternalLogger.Debug("FileTarget: Detected old file in archive. FileName={0}, FileDate={1}, FileDateUtc={2}, CurrentDateUtc={3}, Age={4} days", existingArchiveFile.FileName, existingArchiveFile.Date.ToString("u"), fileDateUtc.ToString("u"), currentDateUtc.ToString("u"), Math.Round(fileAgeDays, 1));
```
Comment (304NotModified) [inline src/NLog/Targets/FileArchiveModes/FileArchiveModeBase.cs]: PS I think you also could do `{2:u}` instead of `ToString("u")`. 

Maybe a bit easier to read, but not very important.

Comment (snakefoot) [reply src/NLog/Targets/FileArchiveModes/FileArchiveModeBase.cs]: I prefer "u" or "o". But find "u" more readable.

File: src/NLog/Targets/FileArchiveModes/FileArchiveModeBase.cs (hunk 1 above)
Comment (snakefoot) [reply src/NLog/Targets/FileArchiveModes/FileArchiveModeBase.cs]: Just removes delegate allocation :)

File: src/NLog/Targets/FileArchiveModes/FileArchiveModeBase.cs (hunk 2 above)
Comment (304NotModified) [reply src/NLog/Targets/FileArchiveModes/FileArchiveModeBase.cs]: I mean,

instead of:

```c#
InternalLogger.Debug("Date={0},", existingArchiveFile.FileName, Date.ToString("u"))
```

A more reabable way would be:

```c#
InternalLogger.Debug("Date={0:u},", existingArchiveFile.FileName, Date)
```

:)

Comment (snakefoot) [reply src/NLog/Targets/FileArchiveModes/FileArchiveModeBase.cs]: Well it still performs boxing etc. But if you really want it then ok.

File: src/NLog/Targets/FileArchiveModes/FileArchiveModeBase.cs (hunk 1 above)
Comment (304NotModified) [reply src/NLog/Targets/FileArchiveModes/FileArchiveModeBase.cs]: I really was under the impression that the "method group" call was better I'm perf (no anonymous method) , but have found a good article on that. Do you have something to read for me?

**Comment by murrays1208049** [comment]
@snakefoot Hello, I have a stackoverflow question [https://stackoverflow.com/questions/73783383/nlog-maxarchivedays-delete-old-folders](https://stackoverflow.com/questions/73783383/nlog-maxarchivedays-delete-old-folders). This question troubled me a long time. Could you help me figure out
